packages/marketplace/test/exchange/Batching.behavior.ts

```diff
@@ -0,0 +1,388 @@
+import {expect} from 'chai';
+import {deployFixtures} from '../fixtures.ts';
+import {loadFixture} from '@nomicfoundation/hardhat-network-helpers';
+import {AssetERC20, AssetERC721, Asset} from '../utils/assets.ts';
+
+import {getSymmetricOrder, OrderDefault, signOrder} from '../utils/order.ts';
+import {ZeroAddress, Contract, Signer} from 'ethers';
+
+// eslint-disable-next-line mocha/no-exports
+export function shouldMatchOrderForBatching() {
+  describe('Exchange MatchOrders for batching', function () {
+    let ExchangeContractAsUser: Contract,
+      OrderValidatorAsAdmin: Contract,
+      ERC20Contract: Contract,
+      ERC721Contract: Contract,
+      maker: Signer,
+      taker: Signer,
+      takerAsset: Asset,
+      totalPayment: number;
+
+    beforeEach(async function () {
+      ({
+        ExchangeContractAsUser,
+        OrderValidatorAsAdmin,
+        ERC20Contract,
+        ERC721Contract,
+        user2: maker,
+        user: taker,
+      } = await loadFixture(deployFixtures));
+    });
```

Comment:
why can't we reuse the vars set in Exchange.test.ts ?

Reply:
scope of the variables is specific and can't be reuse outside the block.

Reply:
Also if we have to use them, alot of variable need to be passed in function call.